### PR TITLE
Fix UnboundLocalError when pycon block is missing ">>>" prompt

### LINKED ISSUE #68: UnboundLocalError: local variable 'indentation' referenced before assignment
```console
$ blacken-docs t.rst 
Traceback (most recent call last):
  File "/tmp/c/venv/bin/blacken-docs", line 8, in <module>
    sys.exit(main())
  File "/tmp/c/venv/lib/python3.8/site-packages/blacken_docs.py", line 210, in main
    retv |= format_file(filename, black_mode, skip_errors=args.skip_errors)
  File "/tmp/c/venv/lib/python3.8/site-packages/blacken_docs.py", line 166, in format_file
    new_contents, errors = format_str(contents, black_mode)
  File "/tmp/c/venv/lib/python3.8/site-packages/blacken_docs.py", line 155, in format_str
    src = RST_PYCON_RE.sub(_rst_pycon_match, src)
  File "/tmp/c/venv/lib/python3.8/site-packages/blacken_docs.py", line 139, in _rst_pycon_match
    code += orig_line[indentation:] + '\n'
UnboundLocalError: local variable 'indentation' referenced before assignment
```

```rst
.. code-block:: pycon

    pass
```

This is clearly bad in that it's not actually a console session (no leading `>>>`), but it would be better to emit a warning and skip the block than to crash with an internal error.

### PR DESCRIPTION
**Pull request by jdufresne**
Fixes #68

### CODE REVIEW

File: tests/blacken_docs_test.py [hunk 1]
```diff
@@ -672,3 +672,27 @@ def test_format_src_rst_pycon_preserves_output_indentation():
     )
     after, _ = blacken_docs.format_str(before, BLACK_MODE)
     assert after == before
+
+
+def test_format_src_rst_pycon_no_prompt():
+    before = (
+        '.. code-block:: pycon\n'
+        '\n'
+        '    pass\n'
```
Comment (asottile) [inline tests/blacken_docs_test.py:681]: I think this should produce an error unless `--skip-errors` is set

Comment (jdufresne) [reply tests/blacken_docs_test.py:681]: Consider the following code block:

```rst
.. code-block:: pycon

    # Comment about next line
    >>> pass
```

Technically, this is not output one would ever see in a Python console as they aren't able to add comments above a prompt. However, even though this is wrong, I'm not sure it is helpful to present an error for the user. This shares the property that the first non-empty line is not prefixed with `>>> `, which I believe is the state you're suggesting should error. WDYT?

Comment (asottile) [reply tests/blacken_docs_test.py:694]: hmmm yeah `doctest` accepts this so it's probably ok?


```console
$ python3 -m doctest t.rst -vv
Trying:
    pass
Expecting nothing
ok
1 items passed all tests:
   1 tests in t.rst
1 tests in 1 items.
1 passed and 0 failed.
Test passed.
```

**Review by asottile — APPROVED** [review]
idk why I took so long to review this, oops

**Pull request merged**